request.go

```diff
@@ -92,7 +92,21 @@ func NewSerializer(opts Options) (encoding.Serializer, error) {
 			return nil, errHealthAndProcedure
 		}
 
-		return opts.ROpts.Encoding.GetHealth()
+		e := opts.ROpts.Encoding
+		if e == encoding.UnspecifiedEncoding {
+			topts, err := loadTransportPeers(opts.TOpts)
```

Comment:
can we try to avoid duplicating this logic. instead, we might want to handle the health request in `withTransportSerializer` or something, and we can just pass in the transport to `GetHealth`?

Reply:
Currently `GetHealth` returns a Serializer, and a serializer is required to create a transport, so that'd be circular. I'll try to find a refactor to avoid having to potentially call `loadTransportPeers` twice and dedupe this.